packages/common/src/graphql/entities.ts

```diff
@@ -1,14 +1,126 @@
+  FieldScalar,
+  GraphQLRelationsType
+} from './types';
+
 export function getAllEntities(schema: GraphQLSchema): GraphQLObjectType[] {
   return Object.entries(schema.getTypeMap())
     .filter(([, node]) => node.astNode?.directives?.find(({name: {value}}) => value === DirectiveName.Entity))
     .map(([, node]) => node)
     .filter(isObjectType);
 }
 
-// TODO: GraphQLNamedType -> EntitySchema
+export function getAllEntitiesRelations(schema: GraphQLSchema): GraphQLModelsRelations{
+  const modelRelations = {models:[],relations:[]} as GraphQLModelsRelations;
+  const entityNameSet = [];
+  const reverseLookupRelations = [];
+
+  const derivedFrom = schema.getDirective('derivedFrom');
+  const entities = Object.entries(schema.getTypeMap())
+      .filter(([, node]) => node.astNode?.directives?.find(({name: {value}}) => value === DirectiveName.Entity))
+      .map(([, node]) => node)
+      .filter(isObjectType);
+
+  for (const entity of entities){
+    entityNameSet.push(entity.name);
+  }
+
+  for (const entity of entities){
+    const newModel: GraphQLModelsType=  {
+      name: entity.name,
+      fields : []
+    }
+    const fields = entity.getFields();
+    for (const k in fields){
+      if (Object.prototype.hasOwnProperty.call(fields, k)) {
+        const typeString = extractType(fields[k].type).toString();
+        //If is a basic scalar type
+        if (Object.values(FieldScalar).includes(typeString)) {
+          newModel.fields.push(packEntityField(typeString, fields[k],'basic'));
+        }
+        // If is a foreign key
+        else if (entityNameSet.includes(typeString) && fields[k].astNode.directives.length ===0){
+          newModel.fields.push(packEntityField(typeString, fields[k],'foreignKey'));
+          modelRelations.relations.push({
+            from:entity.name, //Identity
+            type: 'belongsTo',
+            to:typeString, //Account
+            foreignKey:  `${fields[k].name}Id` //accountId
+          } as GraphQLRelationsType)
+        }
+        // If is derivedFrom
+        else if (entityNameSet.includes(typeString) && fields[k].astNode.directives.length!==0){
+          const directValues = getDirectiveValues(derivedFrom, fields[k].astNode);
+          if (directValues.field){
+            reverseLookupRelations.push({
+              fromEntity:entity.name,  //Account
+              toEntity: typeString,     //Identity
+              type: isListType(isNonNullType(fields[k].type) ? getNullableType(fields[k].type) : fields[k].type)? 'hasMany':'hasOne',
+              foreignKey: `${directValues.field}Id`, //accountId
+              fieldName: fields[k].name, //identity
+            })
+          }
+        }
+      }
+    }
+    modelRelations.models.push(newModel);
+  }
+  handleDerived(reverseLookupRelations, modelRelations.relations);
+  return modelRelations;
+}
+
+
+function packEntityField(typeString:any, field:GraphQLField<any,any>, fieldType?:string) : GraphQLEntityField{
```

Comment:
use enum for fieldType rather than string

Reply:
also fieldType='base' is never checked, that means your abstraction for this method can be improved